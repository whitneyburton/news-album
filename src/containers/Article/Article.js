import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

export class Article extends Component {
  getPath = () => {
    const { article, match } = this.props;
    if (match.path.includes('favorites')) {
      return '/favorites/'
    } else if (article.url.includes('nationalgeographic')) {
      return '/national-geographic/'
    } else if (article.url.includes('ccn')) {
      return '/crypto-coins/'
    } else if (article.url.includes('newscientist')) {
      return '/new-scientist/'
    }
  }
  
  render() {
    const { article } = this.props;
    const title = article.title.toUpperCase();
    return (
      <Link className='Article' to={this.getPath() + article.id}>
          <div className='Article--title-container'>
            <p className='Article--title'>{title}</p>
          </div>
          <img className='Article--image' alt='article' src={article.urlToImage} />
      </Link>
    )
  }
}

Article.propTypes = {
  article: PropTypes.object
}