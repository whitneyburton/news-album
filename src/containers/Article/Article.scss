.Article {
  position: relative;
  width: 400px;
  margin: 14px auto;
  text-decoration: none;
  color: #000;
}

.Article--title-container {
  background-color: #7080907a;
}

.Article--title {
  width: 90%;
  position: absolute;
  z-index: 1;
  color: #fff;
  font-weight: 500;
  letter-spacing: 2px;
  margin: 16px;
  font-size: 2.2rem;
  text-shadow: -3px 4px 5px black;
}

.Article--image {
  width: 100%;
}

@media screen and (max-width: 450px) {
  .Article {
    .Article--title {
      font-size: 1.3rem;
    }
  }
}