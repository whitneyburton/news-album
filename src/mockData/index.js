export const mockNatGeoArticle = {
  author: 'Douglas Main',
  title: 'Why insect populations are plummeting',
  description: 'Sad insects.',
  url:
    'https://www.nationalgeographic.com/animals/2019/02/why-insect-populations-are-plummeting-and-why-it-matters.html',
  urlToImage:
    'https://www.nationalgeographic.com/content/dam/animals/2019/02/insects-in-decline/01-insects-in-decline-nationalgeographic_1353541.jpg',
  publishedAt: '2019-02-19T02:07:15.6368398Z',
  id: 'b',
  isFavorite: false
};

export const mockCryptoCoinsArticle = {
  author: 'Greg Thompson',
  title: 'Ethereum Slayers',
  description: 'They are everywhere.',
  url: 'https://www.ccn.com/ethereum-killers-crypto-blockchain',
  urlToImage:
    'https://www.ccn.com/wp-content/uploads/2019/02/ethereum-crypto-dark-cement-shutterstock.jpg',
  publishedAt: '2019-02-23T14:28:21Z',
  id: 'f',
  isFavorite: false
};

export const mockNewScientistArticle = {
  author: 'Leah Crane',
  title: 'Four stars full of burnt nuclear ash',
  description: 'Scary stars!',
  url:
    'https://www.newscientist.com/article/2194683-four-stars-full-of-burnt-nuclear-ash-are-speeding-through-the-galaxy/',
  urlToImage:
    'https://images.newscientist.com/wp-content/uploads/2019/02/22150921/vennes1hr.jpg',
  publishedAt: '2019-02-22T00:00:00Z',
  id: 'd',
  isFavorite: false
};

export const mockNatGeoArticles = [
  {
    author: 'Erica Johnson',
    title: '20 U.S. state parks',
    description: 'Pretty parks!',
    url:
      'https://www.nationalgeographic.com/travel/destinations/north-america/united-states/state-parks-worth-the-trip.html',
    urlToImage:
      'https://www.nationalgeographic.com/content/dam/travel/2019-digital/state-parks-to-know/02-state-parks-waianapanapa-park-hawaii.jpg',
    publishedAt: '2019-02-19T02:07:16.9805881Z',
    id: 'a',
    isFavorite: false
  },
  {
    author: 'Douglas Main',
    title: 'Why insect populations are plummeting',
    description: 'Sad insects.',
    url:
      'https://www.nationalgeographic.com/animals/2019/02/why-insect-populations-are-plummeting-and-why-it-matters.html',
    urlToImage:
      'https://www.nationalgeographic.com/content/dam/animals/2019/02/insects-in-decline/01-insects-in-decline-nationalgeographic_1353541.jpg',
    publishedAt: '2019-02-19T02:07:15.6368398Z',
    id: 'b',
    isFavorite: false
  }
];

export const mockNatGeoArticlesWithFav = [
  {
    author: 'Erica Johnson',
    title: '20 U.S. state parks',
    description: 'Pretty parks!',
    url:
      'https://www.nationalgeographic.com/travel/destinations/north-america/united-states/state-parks-worth-the-trip.html',
    urlToImage:
      'https://www.nationalgeographic.com/content/dam/travel/2019-digital/state-parks-to-know/02-state-parks-waianapanapa-park-hawaii.jpg',
    publishedAt: '2019-02-19T02:07:16.9805881Z',
    id: 'a',
    isFavorite: false
  },
  {
    author: 'Douglas Main',
    title: 'Why insect populations are plummeting',
    description: 'Sad insects.',
    url:
      'https://www.nationalgeographic.com/animals/2019/02/why-insect-populations-are-plummeting-and-why-it-matters.html',
    urlToImage:
      'https://www.nationalgeographic.com/content/dam/animals/2019/02/insects-in-decline/01-insects-in-decline-nationalgeographic_1353541.jpg',
    publishedAt: '2019-02-19T02:07:15.6368398Z',
    id: 'b',
    isFavorite: true
  }
];

export const mockNewScienceArticles = [
  {
    author: 'Chelsea Whyte',
    title: 'Wolves are spreading in Germany',
    description: 'Wolves everywhere!',
    url:
      'https://www.nationalgeographic.https://www.newscientist.com/article/2194696-wolves-are-spreading-in-germany-by-moving-into-military-land//animals/2019/02/why-insect-populations-are-plummeting-and-why-it-matters.html',
    urlToImage:
      'https://images.newscientist.com/wp-content/uploads/2019/02/22145340/rm7694.jpg',
    publishedAt: '2019-02-23T00:00:00Z',
    id: 'c',
    isFavorite: false
  },
  {
    author: 'Leah Crane',
    title: 'Four stars full of burnt nuclear ash',
    description: 'Scary stars!',
    url:
      'https://www.newscientist.com/article/2194683-four-stars-full-of-burnt-nuclear-ash-are-speeding-through-the-galaxy/',
    urlToImage:
      'https://images.newscientist.com/wp-content/uploads/2019/02/22150921/vennes1hr.jpg',
    publishedAt: '2019-02-22T00:00:00Z',
    id: 'd',
    isFavorite: false
  }
];

export const mockNewScienceArticlesWithFav = [
  {
    author: 'Chelsea Whyte',
    title: 'Wolves are spreading in Germany',
    description: 'Wolves everywhere!',
    url:
      'https://www.nationalgeographic.https://www.newscientist.com/article/2194696-wolves-are-spreading-in-germany-by-moving-into-military-land//animals/2019/02/why-insect-populations-are-plummeting-and-why-it-matters.html',
    urlToImage:
      'https://images.newscientist.com/wp-content/uploads/2019/02/22145340/rm7694.jpg',
    publishedAt: '2019-02-23T00:00:00Z',
    id: 'c',
    isFavorite: false
  },
  {
    author: 'Leah Crane',
    title: 'Four stars full of burnt nuclear ash',
    description: 'Scary stars!',
    url:
      'https://www.newscientist.com/article/2194683-four-stars-full-of-burnt-nuclear-ash-are-speeding-through-the-galaxy/',
    urlToImage:
      'https://images.newscientist.com/wp-content/uploads/2019/02/22150921/vennes1hr.jpg',
    publishedAt: '2019-02-22T00:00:00Z',
    id: 'd',
    isFavorite: true
  }
];

export const mockCryptoCoinsArticles = [
  {
    author: 'Joseph Young',
    title: 'Bitcoins spike',
    description: 'So much money!',
    url:
      'https://www.ccn.com/bitcoin-price-4100-2-hours-analyst-crypto-bullish',
    urlToImage:
      'https://www.ccn.com/wp-content/uploads/2019/02/bitcoin-price-green-chart-bull-shutterstock.jpg',
    publishedAt: '2019-02-23T14:45:23Z',
    id: 'e',
    isFavorite: false
  },
  {
    author: 'Greg Thompson',
    title: 'Ethereum Slayers',
    description: 'They are everywhere.',
    url: 'https://www.ccn.com/ethereum-killers-crypto-blockchain',
    urlToImage:
      'https://www.ccn.com/wp-content/uploads/2019/02/ethereum-crypto-dark-cement-shutterstock.jpg',
    publishedAt: '2019-02-23T14:28:21Z',
    id: 'f',
    isFavorite: false
  }
];

export const mockCryptoCoinsArticlesWithFav = [
  {
    author: 'Joseph Young',
    title: 'Bitcoins spike',
    description: 'So much money!',
    url:
      'https://www.ccn.com/bitcoin-price-4100-2-hours-analyst-crypto-bullish',
    urlToImage:
      'https://www.ccn.com/wp-content/uploads/2019/02/bitcoin-price-green-chart-bull-shutterstock.jpg',
    publishedAt: '2019-02-23T14:45:23Z',
    id: 'e',
    isFavorite: false
  },
  {
    author: 'Greg Thompson',
    title: 'Ethereum Slayers',
    description: 'They are everywhere.',
    url: 'https://www.ccn.com/ethereum-killers-crypto-blockchain',
    urlToImage:
      'https://www.ccn.com/wp-content/uploads/2019/02/ethereum-crypto-dark-cement-shutterstock.jpg',
    publishedAt: '2019-02-23T14:28:21Z',
    id: 'f',
    isFavorite: true
  }
];

export const mockNatGeoArticlesObj = {
  articles: [
    {
      author: 'Erica Johnson',
      title: '20 U.S. state parks',
      description: 'Pretty parks!',
      url:
        'https://www.nationalgeographic.com/travel/destinations/north-america/united-states/state-parks-worth-the-trip.html',
      urlToImage:
        'https://www.nationalgeographic.com/content/dam/travel/2019-digital/state-parks-to-know/02-state-parks-waianapanapa-park-hawaii.jpg',
      publishedAt: '2019-02-19T02:07:16.9805881Z',
      id: 'a',
      isFavorite: false
    },
    {
      author: 'Douglas Main',
      title: 'Why insect populations are plummeting',
      description: 'Sad insects.',
      url:
        'https://www.nationalgeographic.com/animals/2019/02/why-insect-populations-are-plummeting-and-why-it-matters.html',
      urlToImage:
        'https://www.nationalgeographic.com/content/dam/animals/2019/02/insects-in-decline/01-insects-in-decline-nationalgeographic_1353541.jpg',
      publishedAt: '2019-02-19T02:07:15.6368398Z',
      id: 'b',
      isFavorite: false
    }
  ]
};

export const mockNewScienceArticlesObj = {
  articles: [
    {
      author: 'Chelsea Whyte',
      title: 'Wolves are spreading in Germany',
      description: 'Wolves everywhere!',
      url:
        'https://www.nationalgeographic.https://www.newscientist.com/article/2194696-wolves-are-spreading-in-germany-by-moving-into-military-land//animals/2019/02/why-insect-populations-are-plummeting-and-why-it-matters.html',
      urlToImage:
        'https://images.newscientist.com/wp-content/uploads/2019/02/22145340/rm7694.jpg',
      publishedAt: '2019-02-23T00:00:00Z',
      id: 'c',
      isFavorite: false
    },
    {
      author: 'Leah Crane',
      title: 'Four stars full of burnt nuclear ash',
      description: 'Scary stars!',
      url:
        'https://www.newscientist.com/article/2194683-four-stars-full-of-burnt-nuclear-ash-are-speeding-through-the-galaxy/',
      urlToImage:
        'https://images.newscientist.com/wp-content/uploads/2019/02/22150921/vennes1hr.jpg',
      publishedAt: '2019-02-22T00:00:00Z',
      id: 'd',
      isFavorite: false
    }
  ]
};

export const mockCryptoCoinsArticlesObj = {
  articles: [
    {
      author: 'Joseph Young',
      title: 'Bitcoins spike',
      description: 'So much money!',
      url:
        'https://www.ccn.com/bitcoin-price-4100-2-hours-analyst-crypto-bullish',
      urlToImage:
        'https://www.ccn.com/wp-content/uploads/2019/02/bitcoin-price-green-chart-bull-shutterstock.jpg',
      publishedAt: '2019-02-23T14:45:23Z',
      id: 'e',
      isFavorite: false
    },
    {
      author: 'Greg Thompson',
      title: 'Ethereum Slayers',
      description: 'They are everywhere.',
      url: 'https://www.ccn.com/ethereum-killers-crypto-blockchain',
      urlToImage:
        'https://www.ccn.com/wp-content/uploads/2019/02/ethereum-crypto-dark-cement-shutterstock.jpg',
      publishedAt: '2019-02-23T14:28:21Z',
      id: 'f',
      isFavorite: false
    }
  ]
};

export const mockNewSciArticlesStringified = {
  json: {
    articles: [
      {
        author: 'Chelsea Whyte',
        description: 'Wolves everywhere!',
        id: 'c',
        isFavorite: false,
        publishedAt: '2019-02-23T00:00:00Z',
        title: 'Wolves are spreading in Germany',
        url:
          'https://www.nationalgeographic.https://www.newscientist.com/article/2194696-wolves-are-spreading-in-germany-by-moving-into-military-land//animals/2019/02/why-insect-populations-are-plummeting-and-why-it-matters.html',
        urlToImage:
          'https://images.newscientist.com/wp-content/uploads/2019/02/22145340/rm7694.jpg'
      },
      {
        author: 'Leah Crane',
        description: 'Scary stars!',
        id: 'd',
        isFavorite: false,
        publishedAt: '2019-02-22T00:00:00Z',
        title: 'Four stars full of burnt nuclear ash',
        url:
          'https://www.newscientist.com/article/2194683-four-stars-full-of-burnt-nuclear-ash-are-speeding-through-the-galaxy/',
        urlToImage:
          'https://images.newscientist.com/wp-content/uploads/2019/02/22150921/vennes1hr.jpg'
      }
    ]
  },
  ok: true
};
