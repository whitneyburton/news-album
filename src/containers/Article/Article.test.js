import React from 'react';
import { shallow } from 'enzyme';
import { Article } from './Article';
import * as data from '../../mockData';

describe('Article', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Article article={data.mockNatGeoArticle}/>)
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('getPath', () => {
    it('should return a nat geo path', () => {
      const expected = '/national-geographic/';
      const result = wrapper.instance().getPath();
      expect(result).toEqual(expected);
    });

    it('should return a crypto coins path', () => {
      wrapper = shallow(<Article article={data.mockCryptoCoinsArticle} />);
      const expected = '/crypto-coins/';
      const result = wrapper.instance().getPath();
      expect(result).toEqual(expected);
    });

    it('should return a new scientist path', () => {
      wrapper = shallow(<Article article={data.mockNewScientistArticle} />);
      const expected = '/new-scientist/';
      const result = wrapper.instance().getPath();
      expect(result).toEqual(expected);
    });
  });
});