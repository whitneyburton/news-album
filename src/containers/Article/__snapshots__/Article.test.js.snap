// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article should match the snapshot 1`] = `
<Link
  className="Article"
  replace={false}
  to="/national-geographic/undefined"
>
  <div
    className="Article--title-container"
  >
    <p
      className="Article--title"
    >
      LIONS AND TIGERS AND BEARS
    </p>
  </div>
  <img
    alt="article"
    className="Article--image"
    src="www.image.com"
  />
</Link>
`;
